src/scraper.py

```diff
@@ -1,234 +1,245 @@
+def get_game_data(
+    system_id: str,
+    rom_path: str,
+    dev_id: str,
+    dev_password: str,
+    username: str,
+    password: str,
+) -> Any:
     game_url = parse_find_game_url(
         system_id, rom_path, dev_id, dev_password, username, password
     )
     return fetch_data(game_url)
 
 
-def get_user_data(dev_id, dev_password, username, password):
+def get_user_data(dev_id: str, dev_password: str, username: str, password: str) -> Any:
     user_info_url = parse_user_info_url(dev_id, dev_password, username, password)
     return fetch_data(user_info_url)
 
 
-def _fetch_media(medias, properties, regions):
+def _fetch_media(medias: List[dict], properties: dict, regions: List[str]) -> bytes:
     media_type = properties["type"]
     media_height = properties["height"]
     media_width = properties["width"]
 
-    if not is_media_type_valid(media_type):
-        return None
-
+    is_media_type_valid(media_type)
     media_url = find_media_url_by_region(medias, media_type, regions)
-    if media_url:
-        media_url = add_wh_to_media_url(media_url, media_width, media_height)
-        return get(media_url)
-    return None
+    media_url = add_wh_to_media_url(media_url, media_width, media_height)
+    return get(media_url)
 
 
-def fetch_box(game, config):
+def fetch_box(game: dict, config: dict) -> Optional[bytes]:
     medias = game["response"]["jeu"]["medias"]
     regions = config.get("regions", ["us", "ame", "wor"])
-    box = _fetch_media(medias, config["box"], regions)
-    if not box:
-        logger.log_error(f"Error downloading box: {game['response']['jeu']['medias']}")
-        return None
-    return box
+    try:
+        box = _fetch_media(medias, config["box"], regions)
+        return box
+    except exceptions.ScraperError as e:
+        raise exceptions.ScraperError(f"Error downloading box: {e}")
 
 
-def fetch_preview(game, config):
+def fetch_preview(game: dict, config: dict) -> Optional[bytes]:
     medias = game["response"]["jeu"]["medias"]
     regions = config.get("regions", ["us", "ame", "wor"])
-    preview = _fetch_media(medias, config["preview"], regions)
-    if not preview:
-        logger.log_error(
-            f"Error downloading preview: {game['response']['jeu']['medias']}"
-        )
-        return None
-    return preview
+    try:
+        preview = _fetch_media(medias, config["preview"], regions)
+        return preview
+    except exceptions.ScraperError as e:
+        raise exceptions.ScraperError(f"Error downloading preview: {e}")
 
 
-def fetch_synopsis(game, config):
+def fetch_synopsis(game: dict, config: dict) -> Optional[str]:
     synopsis = game["response"]["jeu"].get("synopsis")
     if not synopsis:
         return None
 
     synopsis_lang = config["synopsis"]["lang"]
     synopsis_text = next(
-        (item["text"] for item in synopsis if item["langue"] == synopsis_lang), None
+        (item["text"] for item in synopsis if item.get("langue") == synopsis_lang), None
```

Comment:
Not sure if there is a good reason not to do this or if this will even end up throwing an error if `"text"` doesn't exist, but just something I noticed and felt like it might help if I pointed it out. (Disclaimer: I am not too familiar with python)
```suggestion
        (item.get("text") for item in synopsis if item.get("langue") == synopsis_lang), None
```